src/main/java/net/sf/jabref/gui/openoffice/OpenOfficePanel.java

```diff
@@ -322,6 +332,39 @@ public void actionPerformed(ActionEvent e) {
 
     }
 
+    private void exportEntries() {
+        try {
+            if (style == null) {
+                style = loader.getUsedStyle();
+            } else {
+                style.ensureUpToDate();
+            }
+
+            ooBase.updateSortedReferenceMarks();
+
+            List<BibDatabase> databases = getBaseList();
+            List<String> unresolvedKeys = ooBase.refreshCiteMarkers(databases, style);
+            BibDatabase newDatabase = ooBase.generateDatabase(databases);
+            if (!unresolvedKeys.isEmpty()) {
+                JOptionPane.showMessageDialog(frame,
+                        Localization.lang(
+                                "Your OpenOffice/LibreOffice document references the BibTeX key '%0', which could not be found in your current database.",
+                                unresolvedKeys.get(0)),
+                        Localization.lang("Unable to generate new database"), JOptionPane.ERROR_MESSAGE);
+            }
+
+            Defaults defaults = new Defaults(
+                    BibDatabaseMode.fromPreference(Globals.prefs.getBoolean(JabRefPreferences.BIBLATEX_DEFAULT_MODE)));
+
+            BibDatabaseContext databaseContext = new BibDatabaseContext(newDatabase, defaults);
+            this.frame.addTab(databaseContext, true);
+
+        } catch (Exception e) {
```

Comment:
Another option would be to convert the `Exception` thrown by `com.sun.star.lang.XMultiServiceFactory.createInstance` into a custom `JabRefException` (the class does already exist) and check for that. This  preserves stack trace and message of the original exception and is as simple as:
```
catch(Exception e) {
  throw new JabRefException(e);
}
```
Then you can only deal with `JabRefException` from that point on and avoid accidentally catching another one. It also doesn't clutter the code with handling code. You can do the handling as you did before, just on a more limited type of exception. Would this be an option for the OO/LO part?